### ISSUE TITLE
got overfitting when training?

### ISSUE THREAD

**Issue by Qg2TzVSm** [matthewearl/deep-anpr#39]
Dear all.

I modify the code to recognise Chinese car plate, and successful train the model.  But I can't detect any real car plate, but it worked on test plates which generated by gen.py.  Maybe I generate too many test images in the .test/ folder? How many test images should I generate?

Is the model got overfitting? Can I add some real car plate in the test set images to solve this problem, or add a drop out in full connected layer?

Thanks.
(excuse for my English.)

**Comment by sourabh2k15** [comment]
Hello, I had trained this for Indian number plates and it worked fine. How are you generating the test data , can you provide me an example so I can help out. 

Thanks.

**Comment by zhmd** [comment]
Hi,

If I am understanding correctly, the images generated in ./test folder are used only for test purpose, thus should have nothing to do with the training phase. Therefore, the number of test images you generated should have no impacts on the model. With that being said, I don't think your problem is due to overfitting. In other words, even if you add real car plate to ./test folder it won't improve the model. 

In fact, could you be more specific about how you trained the model on Chinese car plate? If you want it to work with Chinese car plate, you need to re-write the `gen.py` as that's how the real training samples are created. Did you do that?

Thanks,

**Comment by Qg2TzVSm** [comment]
Hello, 
Sorry for so long to reply. I got the weights.npz after a 10 hours training. It could not recognise any real car plate pictures but work well with the training set pictures(test images).
Here is the test images I generated:
<a data-flickr-embed="true"  href="https://www.flickr.com/photos/151048597@N08/34752027261/in/dateposted-public/" title="2"><img src="https://c1.staticflickr.com/5/4268/34752027261_390d7ce0ef_z.jpg" width="640" height="341" alt="2"></a>
Here is a real car plate image:
<a data-flickr-embed="true"  href="https://www.flickr.com/photos/151048597@N08/34497362460/in/dateposted-public/" title="9"><img src="https://c1.staticflickr.com/5/4203/34497362460_cd53a36d5d.jpg" width="400" height="300" alt="9"></a>
Maybe I should do some change to the test images?
Thanks for your help! (excuse for my Englisch again

**Comment by jesiws** [comment]
I'm also trying to modified the code to recognize Chinese car plates, and I find that just because the chars in the generated images are black but in real plate are white that causes the problem. Then I reverse the color of the real test images and find the model works out fine. Hope it may helps you.

By the way, what kind of font are you using to generated images? I think the images you provided above don't look quite like real plates. This may also cause the problem of failure in detection.

**Comment by Qg2TzVSm** [comment]
@jesiws Thanks! I 'll try your advice.
I used different fonts to generate images. Does this improve the ability of the model to identify similar characters? Did you change the size of the test images?
